com.microsoft.mrtk.core/Editor/MRTKProjectValidation.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Editor;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.XR.CoreUtils.Editor;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+    /// <summary>
+    /// A class helping users with validating the configuration of the project for use with MRTK3
+    /// </summary>
+    public static class MRTKProjectValidation
+    {
+        private const string XRProjectValidationSettingsPath = "Project/XR Plug-in Management/Project Validation";
+        private const string DefaultMRTKProfileGuid = "c677e5c4eb85b7849a8da406775c299d";
+        private static readonly BuildTargetGroup[] buildTargetGroups = ((BuildTargetGroup[])Enum.GetValues(typeof(BuildTargetGroup))).Distinct().ToArray();
+        private static readonly Dictionary<BuildTargetGroup, List<BuildValidationRule>> validationRulesDictionary = new Dictionary<BuildTargetGroup, List<BuildValidationRule>>();
+
+        [MenuItem("Mixed Reality/MRTK3/Utilities/Project Validation", priority = 0)]
+        private static void MenuItem()
+        {
+            SettingsService.OpenProjectSettings(XRProjectValidationSettingsPath);
+        }
+
+        [InitializeOnLoadMethod]
+        private static void MRTKProjectValidationCheck()
+        {
+            AddMRTKCoreValidationRules();
+            // Add a delay to wait for all rules to be added (potentially from other MRTK scripts registered with InitializeOnLoadMethod)
+            EditorApplication.delayCall += AddMRTKValidationRules;
+        }
+
+        private static void AddMRTKValidationRules()
+        {
+            foreach (var buildTargetGroup in validationRulesDictionary.Keys)
+            {
+                BuildValidator.AddRules(buildTargetGroup, validationRulesDictionary[buildTargetGroup]);
+            }
+        }
+
+        private static void AddMRTKCoreValidationRules()
+        {
+            List<BuildValidationRule> mrtkCoreTargetIndependentRules = new List<BuildValidationRule>
+            {
+                // Always ensure the standalone target has a profile assigned no matter what target is being targeted
+                GenerateProfileRule(BuildTargetGroup.Standalone),
```

Comment:
Should there be a profile rule for every non-server platform? That way, the developer knows they are missing the Android, for example, profile...

Reply:
Yes, we have those rules on line 64.